src/Symfony/Component/HttpClient/HttplugClient.php

```diff
@@ -229,9 +229,14 @@ private function sendPsr7Request(RequestInterface $request, ?bool $buffer = null
                 $body->seek(0);
             }
 
+            $headers = $request->getHeaders();
+            if (($size = $body->getSize()) !== null) {
```

Comment:
Just in case:
```suggestion
            if (!isset($headers['Content-Length']) && !isset($headers['content-length']) && 0 <= $size = $body->getSize() ?? -1) {
```

Reply:
I used `$request->hasHeader('content-length')` instead of the `isset` check to make sure any weird capitalization works.
`$stream->getSize()` can't return -1 when implemented correctly according to the PSR, but the extra check won't hurt.